docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+
+The literal passed as input to the suffix function must be immediately followed by the name of the suffix.
+The two must be separated by whitespace, unless it's a string, unicode or hexadecimal string literal,
+in which case the whitespace is optional.
+
+This call syntax supports only a single literal argument.
+Variables or expressions (even as simple as wrapping the literal in parentheses) are not allowed.
+Suffix functions defined with two parameters are also invoked with one literal - the
+:ref:`decomposition<fractional_decomposition>` of the literal into two values is performed implicitly
+by the compiler.
+
+.. note::
+    There are no negative number literals in Solidity.
+    A literal with a minus sign is an expression.
+    ``-123 suffix`` is equivalent to ``-(123 suffix)``, so ``suffix`` does not receive ``-123`` as input.
+    The argument is instead ``123`` and the negation is applied to the returned value.
+
+.. note::
+    String concatenation produces a single literal at compilation time and therefore is not treated
+    as an expression.
+    This means that e.g. ``"abc" "def" suffix`` is a valid suffix call.
+
+.. index:: ! literal suffix; function call syntax, overload
+
+Function Call Syntax
+^^^^^^^^^^^^^^^^^^^^
+
+Suffix definitions are in all respects valid free functions, and this includes the ability to call
+them directly.
+This makes it possible to call such functions with arguments which are not literals.
+
+.. code-block:: solidity
+
+    suffix(42);
+    suffix(123, 2);
+    suffix("abc");
+    suffix(hex"12ff");
+    suffix(true);
+
+Note that the fractional decomposition is not performed for this kind of call - two-argument
+suffix functions must be explicitly called with two arguments.
+
+Regardless of the call syntax used and in contrast to applying a denomination, the result of the
+call is itself not considered a literal.
+As a consequence, it cannot be used as input of another suffix call, and calculations on it are performed
+within its type rather than in arbitrary precision (as is the case with calculations on rational number
+literals).
+
+.. note::
+    Function call syntax is the only way to pass a negative integer value into a suffix function
+    that has a parameter of a signed integer type.
+
+.. note::
+    As all free functions, suffix definitions can be :ref:`overloaded<overload-function>`.
+    Overloaded suffixes, however, cannot be invoked using the suffix call syntax.
+
+.. index:: ! fractional decomposition
+.. _fractional_decomposition:
+
+Fractional Decomposition
+------------------------
+
+To allow defining suffixes that work with fractional literals, like ``1.23``, the language allows
+a special form of a suffix definition.
+Such a suffix can be considered a more general form of a suffix taking a single integer argument.
+
+A single-parameter suffix can be applied only to those rational number literals, which represent
+integers.
+Let us consider the following suffix definition:
```

Comment:
```suggestion
To allow defining suffixes that work with fractional literals, like ``1.23``, the language allows
a special form of a suffix definition.
Such a suffix can be considered a more general form of a suffix, taking a single integer argument.

A single-parameter suffix can be applied only to those rational number literals which represent
integers.
Let us consider the following suffix definition:
```

Reply:
> more general form of a suffix, taking a single integer argument.

No, that's what I meant. The two arg one is a more general form of the single arg suffix. Adding a comma changes the meaning.